src/app_service/service/keycardV2/dto.nim

```diff
@@ -0,0 +1,169 @@
+import json, strutils
+include ../../common/json_utils
+
+
+
+type StateString* = enum
+  UnknownReaderState = "unknown"
+  NoPCSC = "no-pcsc"
+  InternalError = "internal-error"
+  WaitingForReader = "waiting-for-reader"
+  WaitingForCard = "waiting-for-card"
+  ConnectingCard = "connecting-card"
+  ConnectionError = "connection-error" # NOTE: Perhaps a good place for retry
+  NotKeycard = "not-keycard"
+  PairingError = "pairing-error"
+  EmptyKeycard = "empty-keycard"
+  NoAvailablePairingSlots = "no-available-pairing-slots"
+  BlockedPIN = "blocked-pin" # PIN remaining attempts == 0
+  BlockedPUK = "blocked-puk" # PUK remaining attempts == 0
+  FactoryResetting = "factory-resetting"
+  Ready = "ready"
+  Authorized = "authorized"
+
+type KeycardState* = enum
+  UnknownReaderState = -1,
+  NoPCSCService,
+  PluginReader,
+  InsertKeycard,
+  ReadingKeycard,
+  WrongKeycard,
+  NotKeycard,
+  MaxPairingSlotsReached,
+  Locked,
+  NotEmpty,
+  Empty,
+  Authorized
```

Comment:
Not related to your PR, but FYI I've described the missing states here: 
- https://github.com/status-im/status-desktop/issues/17140